Resources/Prototypes/Entities/Structures/Doors/Turnstiles/turnstiles.yml

```diff
@@ -0,0 +1,78 @@
+﻿- type: entity
+  id: Turnstile
+  parent: BaseStructure
+  name: turnstile
+  description: One-way access - control the flow of foot traffic.
+  components:
+  - type: MeleeSound
+    soundGroups:
+      Brute:
+        path:
+          "/Audio/Weapons/smash.ogg"
+  - type: InteractionOutline
+  - type: Sprite
+    sprite: Structures/Doors/turnstile.rsi
+    noRot: true
+    drawdepth: Mobs
+    state: idle
+  - type: AnimationPlayer
```

Comment:
half of these components are probably inherited from structure like physics and interactionoutline do check it